### ISSUE TITLE
Foreign key field is not being set when using a nested create for a one-to-one relation

### ISSUE THREAD

**Issue by jwld** [prisma/prisma#3081]
## Bug description

Foreign key value is not being set when using a nested create for a one-to-one relation.

## How to reproduce

1. Create the following schema:
```
datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

generator client {
  provider = "prisma-client-js"
}

model Company {
  id       String    @default(cuid()) @id
  payments Payment[]
}

model Visit {
  id      String   @default(cuid()) @id
  payment Payment?
}

model Payment {
  id        String  @default(cuid()) @id
  company   Company @relation(fields: [companyId], references: [id])
  companyId String
  visit     Visit?  @relation(fields: [visitId], references: [id])
  visitId   String?
}
```

2. Create a company (need the id for the next step)

3. Run the following script
```
const { PrismaClient } = require('@prisma/client')

const prisma = new PrismaClient()

const main = async () => {
  const newVisit = await prisma.visit.create({ data: {} })

  await prisma.visit.update({
    data: {
      payment: {
        create: {
          company: { connect: { id: '{{ company id here }}' }}
        }
      }
    },
    where: { id: newVisit.id }
  })
}

main()
  .then(() => console.log('Done 🎉'))
  .catch(err => console.log(err))
  .finally(() => prisma.disconnect())
```

4. Check the database - `payment.companyId` will be set, however `payment.visitId` is still `NULL`, despite the payment being created within a visit update. There's no way to connect that payment to visit, since the nested create does not allow a direct visit connection (since it's part of a visit update).

Without the company connection it works, so it's perhaps something to do with connecting to multiple parents?

## Expected behavior

Payments created within an update visit mutation should be connected to the visit.

## Environment & setup

- OS: Mac OS
- Database: PostgreSQL
- Prisma version: 2.2.2
- Node.js version: 14.5.0

**Comment by jwld** [comment]
Note also it does work if changed to a one-many relationship, where a visit has many payments.

**Comment by divyenduz** [comment]
I can reproduce this with `2.4.0-dev.14`

**Comment by jwld** [comment]
@divyenduz Just hit this again in another situation. It's a major pain in the ass because it means we have to run separate transactions, which is riskier of course. Any chance of a fix anytime soon?

**Comment by divyenduz** [comment]
@jwld Thanks for raising this. It is marked as a candidate so it might get picked up soon.

**Comment by jwld** [comment]
Appreciate the response, thanks!

**Comment by jwld** [comment]
Have hit this in a third situation, so still an issue in `2.8`. As usual it requires some funky workarounds. Apologies for spamming but just to indicate that this is quite a pervasive issue for us.

**Comment by dpetrick** [comment]
A fix will be available in version 2.11.
